### ISSUE TITLE
Double tap speed

### ISSUE THREAD

**Issue by JibstaMan** [UltimateHackingKeyboard/agent#613]
I'd like to have a setting to change the double tapping speed required to lock any of the layers, but especially the mouse layer. The mouse layer is tapped with the pinky finger and with the current configuration speed, I find it hard to tap it fast enough to have it lock the layer.

I'm not sure whether this is already (partly) implemented in #501, but I can't seem to find the setting in Agent (1.2.0).

**Comment by mondalaci** [comment]
I agree that this would be useful. There are actually two relevant values. The first is the timeout between the two taps. The second is how long the second tap stays pressed. The second tap should be released quickly to avoid accidental layer locking.

It'd probably be also useful to set individual layer switcher keys lockable vs unlockable via an additional option. Mod is far easier to lock accidentally due to its frequent use, and some users may don't want to lock some layer switcher keys at all.

**Comment by mondalaci** [comment]
@josh-hernandez-exe The double tap speed is not adjustable yet. We'll make it happen later.

**Comment by josh-hernandez-exe** [comment]
@mondalaci, I apologize for my poor choice of words. I was referring to the locking of the layer, not the configurability of the double tap speed. It was a minor personal revolution that I can "lock" a layer. That you for your patience. I apologize for what might seem like trivial comments from myself.

**Comment by farzher** [comment]
just want to mention that `tap to toggle / hold to activate` could be a useful option if it's easy to add.

it's the same as toggle layer, but if i only need to use it for a few seconds i can just hold, then let go, instead of needing to click the key again to turn it off

**Comment by farzher** [comment]
i'm using "toggle the mouse layer by tapping this key"
sometimes i only want to use the mouse for a few seconds.
it'd be nice if when i hold the mouse key instead of tapping it, that when i let go it would disable the mouse layer. (it lets me press the button once instead of twice)

when you tap the key it toggles, when you hold the key it activates/deactivates.

**Comment by mondalaci** [comment]
@farzher Now I see what you mean. We can implement this if there's a wide demand, but I'd hold on until that.

**Comment by farzher** [comment]
yeah, not important. but it seems pretty good.
the current double tap feature is nice but a little too much effort

**Comment by farzher** [comment]
it's not currently possible to disable the double tap, right? it's happening on accident, when i'm being indecisive

**Comment by mondalaci** [comment]
@farzher Not yet, but this is almost ready and will be possible soon. You'll be able to set it a per key basis.

**Comment by jakedowns** [comment]
any update on this one? i run into accidental mod-lock almost daily

**Comment by jakedowns** [comment]
@mondalaci Thanks so much, that worked perfectly
